apps/src/code-studio/initApp/project.js

```diff
@@ -1138,13 +1141,15 @@ var projects = (module.exports = {
           if (err) {
             if (err.message.includes('httpStatusCode: 401')) {
               this.showSaveError_();
+              callback(err);
```

Comment:
I don't think this and the other `callback(err)` are needed if you set `hasProjectChanged` to `false` within this function.